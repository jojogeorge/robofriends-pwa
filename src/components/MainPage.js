import React, { Component } from 'react';
import CardList from './CardList';
import SearchBox from './SearchBox';
import Scroll from './Scroll';
import ErrorBoundry from './ErrorBoundry';

import './MainPage.css';



class MainPage extends Component {
  componentDidMount() {
    this.props.onRequestRobots();
  }

  filterRobots = () =>{
    return this.props.robots.filter(robot => {
      return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase());
    })
  }

  render() {
    const { robots, onSearchChange, isPending } = this.props;
    
    return (
      <div className='tc'>
        <h1 className='f1'>RoboFriends</h1>
        <SearchBox searchChange={onSearchChange}/>
        <Scroll>
          { isPending ? <h1>Loading</h1> :
            <ErrorBoundry>
              <CardList robots={this.filterRobots()} />
            </ErrorBoundry>
          }
        </Scroll>
      </div>
    );
  }
}

// action done from mapDispatchToProps will channge state from mapStateToProps
export default MainPage
