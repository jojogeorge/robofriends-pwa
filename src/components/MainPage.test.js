import {shallow} from 'enzyme'
import React from 'react';
import renderer from 'react-test-renderer'
import MainPage from './MainPage'


describe('MainPage Component',()=> {
    let wrapper;
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots:[],
        searchField: '',
        isPending: false,

    };
    beforeEach(() => {
        wrapper = shallow(<MainPage {...mockProps} />)
    });
   
    it('expect MainPage compoent to be renedered', ()=>{
        expect(wrapper.length).toEqual(1);
    });
        
    it('expect MainPage compoent to be renedered with snapshot', ()=>{     
        const tree = renderer.create(<MainPage {...mockProps}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('it filters robots correctly',()=>{
        const mockProps2 = {
            onRequestRobots: jest.fn(),
            robots:[{
                id:3,
                name: 'john',
                email: 'john@gmail.com'
            }],
            searchField: 'john',
            isPending: false,

        };
        const wrapper2 =  shallow(<MainPage {...mockProps2}/>);
        const filteredRobots = [{
            id:3,
            name: 'john',
            email: 'john@gmail.com'
        }];
        expect(wrapper2.instance().filterRobots()).toEqual(filteredRobots);
    });
    it('it filters robots correctly for wrong search',()=>{
        const mockProps3 = {
            onRequestRobots: jest.fn(),
            robots:[{
                id:3,
                name: 'john',
                email: 'john@gmail.com'
            }],
            searchField: 'a',
            isPending: false,

        };
        const wrapper2 =  shallow(<MainPage {...mockProps3}/>);
        expect(wrapper2.instance().filterRobots()).toEqual([]);
    });

});
