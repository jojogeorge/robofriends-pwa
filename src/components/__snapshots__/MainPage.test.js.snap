// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage Component expect MainPage compoent to be renedered with snapshot 1`] = `
<div
  className="tc"
>
  <h1
    className="f1"
  >
    RoboFriends
  </h1>
  <div
    className="pa2"
  >
    <input
      aria-label=" Search Robots"
      className="pa3 ba b--green bg-lightest-blue"
      placeholder="search robots"
      type="search"
    />
    <button
      id="counter"
      onClick={[Function]}
    >
      count:
      0
    </button>
  </div>
  <div
    style={
      Object {
        "border": "5px solid black",
        "height": "800px",
        "overflow": "scroll",
      }
    }
  >
    <div />
  </div>
</div>
`;
